api/paidAction/downZap.js

```diff
@@ -0,0 +1,44 @@
+export const anonable = false
+export const supportsPessimism = false
+export const supportsOptimism = true
+
+export async function getCost ({ sats }) {
+  return BigInt(sats) * BigInt(1000)
+}
+
+export async function perform ({ invoiceId, sats, id: itemId }, { me, cost, models, tx }) {
+  itemId = parseInt(itemId)
+
+  let invoiceData = {}
+  if (invoiceId) {
+    invoiceData = { invoiceId, invoiceActionState: 'PENDING' }
+  }
+
+  const itemAct = await tx.itemAct.create({
+    data: { msats: cost, itemId, userId: me.id, act: 'DONT_LIKE_THIS', ...invoiceData }
+  })
+
+  const [{ path }] = await tx.$queryRaw`SELECT ltree2text(path) as path FROM "Item" WHERE id = ${itemId}`
+  return { id: itemId, sats, act: 'DONT_LIKE_THIS', path, actId: itemAct.id }
+}
```

Comment:
It's probably not a big deal performance-wise since it's just an index lookup but what I meant was that in components/item-act.js, the `modifyActCache` function does not update any ancestors for downzaps:

```javascript
  if (act === 'TIP') {
    // update all ancestors
    path.split('.').forEach(aId => {
      if (Number(aId) === Number(id)) return
      cache.modify({
        id: `Item:${aId}`,
        fields: {
          commentSats (existingCommentSats = 0) {
            return existingCommentSats + sats
          }
        }
      })
    })
  }
```

So this `path` query seems unnecessary since this action is only for downzaps.

Just want to make sure we're on the same page and I didn't miss something

Reply:
Ah I see what you mean! I like the symmetry between item acts all returning the same stuff, but you're right.